src/plugins/intel_gna/frontend/layer_quantizer.hpp

```diff
@@ -105,615 +97,374 @@ struct FakeQuantizeParams {
         break;
     default:
         THROW_GNA_EXCEPTION << "FP32 to " << precision << " not supported";
     }
     return result_ptr;
 }
 
-template <class T, class... Args>
-InferenceEngine::Blob::Ptr make_custom_blob(Args&&... args) {
-    return InferenceEngine::make_shared_blob<T>(InferenceEngine::Precision::fromType<T>(), std::forward<Args>(args)...);
-}
-
-template <class T>
-InferenceEngine::Blob::Ptr make_custom_blob(InferenceEngine::Layout layout, InferenceEngine::SizeVector size) {
-    return InferenceEngine::make_shared_blob<T>(
-        InferenceEngine::TensorDesc(InferenceEngine::Precision::fromType<T>(), size, layout));
-}
-
-template<class QuantDesc, class QuantFunc>
-inline void quantizeWeightsBiases(const QuantDesc & quantDesc,
-                                  InferenceEngine::WeightableLayer *wl,
-                                  const QuantFunc &fnc,
-                                  bool isDiagonal = false) {  // for diagonal layer number of weights and biases significatly smaller
-    // for quantized weights
-    auto intWeights =
-        make_custom_blob<typename QuantDesc::WeightsPrecision>(InferenceEngine::C, InferenceEngine::SizeVector({wl->_weights->size()}));
-    intWeights->allocate();
-    if (intWeights->buffer() == nullptr) {
-        IE_THROW(NotAllocated)
-                << "[GNAPlugin] in function " << __PRETTY_FUNCTION__<< ": "
-                << "cannot copy weights for layer :"<< wl->name << " of size" << intWeights->byteSize();
+static size_t GetBiasSizeForLayer(InferenceEngine::WeightableLayer* wl) {
+    if (wl->_biases) {
+        return wl->_biases->size();
+    } else if (LayerInfo(*wl).isConvolution()) {
+        // calculating biases len using outdata dims: biases number should be equal to output channels number
+        size_t size = InferenceEngine::GetDataDimSize(wl->outData.front(), InferenceEngine::DataDimName::C);
+        return size;
+    } else {
+        // calculating biases size using outData dimensions
+        const auto& dims = wl->outData.front()->getDims();
+        const auto& o_idx = wl->outData[0]->getDims().size() - 1;
+        return dims[o_idx];
     }
+}
 
-    int oIdx = wl->outData[0]->getDims().size() - 1;
-    int iIdx = wl->insData[0].lock().get()->getDims().size() - 1;
-
-    auto getBiasSizeForLayer = [&oIdx](InferenceEngine::WeightableLayer *wl) {
-        if (wl->_biases) {
-            return wl->_biases->size();
+static std::pair<size_t, size_t> GetNumRowsColumns(InferenceEngine::WeightableLayer* wl) {
+    size_t num_rows;
+    size_t num_columns;
+    const auto& o_idx = wl->outData[0]->getDims().size() - 1;
+    const auto& i_idx = wl->insData[0].lock().get()->getDims().size() - 1;
+
+    if (LayerInfo(*wl).isScaleShift()) {
+        num_columns = 1;
+        num_rows = wl->_weights->size();
+    } else if (LayerInfo(*wl).isConvolution() || LayerInfo(*wl).isConvolutionFilter()) {
```

Comment:
```suggestion
    } else if (LayerInfo(wl).isConvolution() || LayerInfo(wl).isConvolutionFilter()) {
```
// I know this is legacy code, but please
// please try to use in context of one method (if it is possible) one way of retrieving Layer info.